### ISSUE TITLE
`Icons` should be exempted from `Manifest-Metadata-Consistency` check

### ISSUE THREAD

**Issue by mdanish-kh** [microsoft/winget-pkgs#177487]
I'm assuming `Icons:` are injected somewhere in the publish package step as they are not accepted in the community repository. In that case, `Manifest-Metadata-Consistency` check should not check against `Icons:` field. It causes confusion for the PR author as they can't figure out why the bot is reporting a warning since they can't see Icons field in a previous manifest as well.

Some PR examples:

- https://github.com/microsoft/winget-pkgs/pull/177468
- https://github.com/microsoft/winget-pkgs/pull/177463
- https://github.com/microsoft/winget-pkgs/pull/177421

**Comment by mdanish-kh** [comment]
[Policy] Issue-Bug
[Policy] Area-Bots

**Comment by denelon** [comment]
Good catch! Do you have any instances where the PR author said anything in comments?

**Comment by denelon** [comment]
> I'm assuming Icons: are injected somewhere in the publish package step

Yes, we didn't want to create "extra" traffic for publishers or "extra" cost for the data transfer. They are captured during validation so we can store them on the CDN and add the metadata to the manifest based on the "installed" icon.

**Comment by mdanish-kh** [comment]
I have not yet. That's probably because all the current example PRs are "automated" PRs, but I imagine a newcomer may feel confused if they contribute to a relevant package. You can filter the existing PRs where this check is applied incorrectly with this GitHub filter. Checks for "in:comments Icons"

https://github.com/microsoft/winget-pkgs/pulls?page=11&q=is%3Apr+Icons+label%3AManifest-Metadata-Consistency+in%3Acomments+Icons

**Comment by denelon** [comment]
OK. The engineering team filed an internal bug to address it.

**Comment by Alphrag** [comment]
Hi, just to chime in as an author of a package that has an automated build to submit updated manifests, I have been quite surprised by the new warning that has recently started appearing, but decided not to pay to much attention in the first place because the reviewers were accepting the PRs nonetheless.
When I got myself to look into this recently because it was bothering me, I could not find the place of that `Icon` property initially since this is not part of the list of fields that should appear in the manifest in v1.6 [according to the documentation](https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.6.0/defaultLocale.md), though it is indeed present in the [JSON schema of the manifest](https://github.com/microsoft/winget-cli/blob/1b251584211e344c756b8bad10f71db57019aa39/schemas/JSON/manifests/v1.6.0/manifest.defaultLocale.1.6.0.json#L54-L111).

Should we start including this property from now on into the manifest to be in line with the its "required" value in the schema 1.6, or is it something that is going to be left out, and that warning is only going to be temporary? As long as it is clearly laid out and made explicit, making such an addition to the manifests of packages should not be too troublesome, but knowing exactly what is the path we should take to help you keep this list of packages as consistent as possible would be helpful. 
Thanks.

**Comment by mdanish-kh** [comment]
@Alphrag - As mentioned by @denelon above, icons are automatically loaded into the manifest by the CI pipelines. The community repository does not accept Icons in manifest PRs. The "warning" you see is a bug in the validation step.

**Comment by DuckDuckStudio** [comment]
Guys, any progress on fixing this issue? I've been having this issue with [many of my recent PRs](https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+label%3AManifest-Metadata-Consistency+in%3Acomments+Icons+author%3ADuckDuckStudio) ...

**Comment by drshtmstry** [comment]
#210571 
![Image](https://github.com/user-attachments/assets/26783810-0b80-4866-ab85-98950912a0a6)
